bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/dto/AlarmSensorUnitDataDeserializer.java

```diff
@@ -0,0 +1,119 @@
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link AlarmSensorUnitDataDeserializer} implements AlarmSensorUnitData unit
+ * Json De-serializer.
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@NonNullByDefault
+public class AlarmSensorUnitDataDeserializer implements JsonDeserializer<TouchWandUnitDataAlarmSensor> {
+
+    @Override
+    public TouchWandUnitDataAlarmSensor deserialize(@Nullable JsonElement json, @Nullable Type typeOfT,
+            @Nullable JsonDeserializationContext context) throws JsonParseException {
+
+        TouchWandUnitDataAlarmSensor touchWandUnitDataAlarmSensor = new TouchWandUnitDataAlarmSensor();
+
+        JsonElement myJson = json;
+        if (myJson != null) {
+            JsonObject jsonObject = myJson.getAsJsonObject();
+            touchWandUnitDataAlarmSensor.setId(jsonObject.get("id").getAsInt());
+            touchWandUnitDataAlarmSensor.setName(jsonObject.get("name").getAsString());
+            touchWandUnitDataAlarmSensor.setConnectivity(jsonObject.get("connectivity").getAsString());
+            touchWandUnitDataAlarmSensor.setType(jsonObject.get("type").getAsString());
+            touchWandUnitDataAlarmSensor.setHasBattery(jsonObject.get("hasBattery").getAsBoolean());
+            JsonElement powerMeterElement = jsonObject.get("hasPowerMeter");
+            if (!powerMeterElement.isJsonNull()) {
+                touchWandUnitDataAlarmSensor.setHasPowerMeter(powerMeterElement.getAsBoolean());
+            } else {
+                touchWandUnitDataAlarmSensor.setHasPowerMeter(false);
+            }
+
+            if (!jsonObject.get("status").isJsonNull()) { // Sometimes status in null
+                touchWandUnitDataAlarmSensor.setStatus(jsonObject.get("status").getAsString());
+            }
+
+            GsonBuilder builder = new GsonBuilder();
+            JsonObject currentStatusObj = builder.create().fromJson(jsonObject.get("currStatus").getAsJsonObject(),
+                    JsonObject.class);
+
+            TouchWandAlarmSensorCurrentStatus touchWandUnitDataAlarmSensorCurrentStatus = touchWandUnitDataAlarmSensor
+                    .getCurrStatus();
+
+            for (Entry<String, JsonElement> entry : currentStatusObj.entrySet()) {
+                String key = entry.getKey();
+                String splits[] = key.split("_"); // the key is xxxx_n where xxx is sensor type and n is
+                String keyName = splits[0];
+                int index = 0;
+
+                if (splits.length > 1 && !splits[1].isEmpty()) {
+                    try {
+                        index = Integer.parseInt(splits[1]);
+                    } catch (final NumberFormatException e) {
+                        index = 0;
+                    }
+                }
+
+                final Gson gson = new Gson();
```

Comment:
Why are you creating a new Gson instance each time? Please create your Gson instance statically and make it a static final field.